src/app/hooks/useClickTrackerHandler/index.jsx

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable no-console */
+import { useContext, useCallback, useState } from 'react';
+import path from 'ramda/src/path';
+
+import { sendEventBeacon } from '#containers/ATIAnalytics/beacon/index';
+import { isValidClick } from './clickTypes';
+import { EventTrackingContext } from '#app/contexts/EventTrackingContext';
+import { ServiceContext } from '#contexts/ServiceContext';
+import useToggle from '#hooks/useToggle';
+
+const EVENT_TYPE = 'click';
+
+const useClickTrackerHandler = (props = {}) => {
+  const componentName = path(['componentName'], props);
+  const href = path(['href'], props);
+  const format = path(['format'], props);
+
+  const { enabled: eventTrackingIsEnabled } = useToggle('eventTracking');
+  const [clicked, setClicked] = useState(false);
+  const {
+    campaignName,
+    pageIdentifier,
+    platform,
+    statsDestination,
+  } = useContext(EventTrackingContext);
+  const { service } = useContext(ServiceContext);
+
+  return useCallback(
+    event => {
+      if (eventTrackingIsEnabled && !clicked && isValidClick(event)) {
+        setClicked(true);
+
+        const shouldSendEvent = [
+          campaignName,
+          componentName,
+          pageIdentifier,
+          platform,
+          service,
+          statsDestination,
+        ].every(Boolean);
+
+        if (shouldSendEvent) {
+          const nextPageUrl = href || event.target.href;
+
+          event.stopPropagation();
+          event.preventDefault();
+
+          try {
+            sendEventBeacon({
+              type: EVENT_TYPE,
+              campaignName,
+              componentName,
+              format,
+              pageIdentifier,
+              platform,
+              service,
+              statsDestination,
+            });
+          } finally {
+            if (nextPageUrl) {
+              window.location.assign(nextPageUrl);
```

Comment:
is this a security risk? should we use react-router-dom redirect component?

Reply:
@HarryVerhoef are you able to look into this?

Reply:
@jroebu14 Yeah I can try and find something 👍

Reply:
@andrewscfc @jroebu14 I think using react-router we'd only be able to redirect to a relative url, and so wouldn't be able to do an external redirect.

Reply:
Do we have any need to support external links? I'd say would be good to use react router if not, if we do I guess stick with existing approach?